packages/lesswrong/components/vulcan-forms/Form.tsx

```diff
@@ -986,6 +986,10 @@ class Form<T extends DbObject> extends Component<any,any> {
     // complete the data with values from custom components
     // note: it follows the same logic as SmartForm's getDocument method
     let data = this.getData({ addExtraFields: false });
+    
+    if (isPromise(data)) {
+      data = await data
```

Comment:
Would be better to fold this await into the inside of `this.getData`. Not merge-blocking tho.